src/js/main.js

```diff
@@ -334,6 +343,35 @@ Site.Map = {
 
   },
 
+  showPattern: function() {
+    var _this = this;
+    var patternStyle = parseInt(Math.random() * 2);
+
+    if (patternStyle === 1) {
+      $('#background-pattern-holder').addClass('fill-path');
+    } else {
+      $('#background-pattern-holder').addClass('stroke-path');
+    }
+
+    _this.currentPattern = Math.floor(Math.random() * (_this.patternMax - _this.patternMin + 1)) + _this.patternMin;
+
+    $('svg#pattern-' + _this.currentPattern).addClass('show');
+  },
+
+  changePattern: function() {
+    var _this = this;
+
+    $('.map-pattern').removeClass('show');
+
+    if (_this.currentPattern >= _this.patternMax) {
+      _this.currentPattern = _this.patternMin;
+    } else {
+      _this.currentPattern++;
+    }
+
+    $('svg#pattern-' + _this.currentPattern).addClass('show');
```

Comment:
you could also cache each of the pattern objects in an array and use the index of the array rather than the string addition style like this